### ISSUE TITLE
vcpkg port and CMake build by KDE

### ISSUE THREAD

**Issue by wrobelda** [libofx/libofx#46]
Hi,

I am one of the KMyMoney maintainers, and I took upon myself to add a port to each of our dependencies to vcpkg. It's been going for a year and I had a pretty good progress so far, with some of the common dependencies of KMyMoney and GnuCash already ported in (see https://github.com/pulls?q=is%3Apr+author%3Awrobelda+archived%3Afalse+is%3Aclosed+repo%3Amicrosoft%2Fvcpkg+).

I just added a [PR](https://github.com/microsoft/vcpkg/pull/20391) for OpenSP, and I am working on LibOFX port. KDE already has a ["blueprint" for it ](https://invent.kde.org/packaging/craft-blueprints-kde/-/blob/master/libs/libofx/libofx.py), which used to work for all platforms, including MSVC – which I also recently updated it to 0.10.2 and only today realized I messed up the Windows build. This is because not all of the old patches apply. Some are no longer needed, some need adjusting. 

I know as well that you had contemplated migration to vcpkg a while ago. Which brings me here: I wish we could find a common ground, and upstream those patches, especially those that touch pkg-config or add the CMakeLists.txt ([(1](https://invent.kde.org/packaging/craft-blueprints-kde/-/blob/2a47ef6c3ef1b1ad1aa3695080ea89cd86c4a6ee/libs/libofx/libofx-0.9.13-20180505-1.diff), [2](https://invent.kde.org/packaging/craft-blueprints-kde/-/blob/2a47ef6c3ef1b1ad1aa3695080ea89cd86c4a6ee/libs/libofx/libofx-0.9.13-20180505-5.diff)). This will reduce the fragmentation and inevitably reduce the time we all spend on supporting our respective build systems. 

FYI, vcpkg currently defaults to static builds on macOS and Linux, so that adds some complexity to the migration process, but effectively helps those builds to be  more robust and fool-proof. 

And BTW, I have a personal list of other tasks left in regards to getting LibOFX/AqBanking upstreamed to vcpkg: https://invent.kde.org/wrobelda/vcpkg/-/boards. On top of these, you would want to add the `Gtk` feature to at least gwenhywfar.

**Comment by wrobelda** [comment]
P.S. I pushed the WIP `libofx` vcpkg port here: https://github.com/microsoft/vcpkg/pull/20426

**Comment by wrobelda** [comment]
@cstim it's been 3 weeks since I originally posted this, I would appreciate some feedback so that I can move forward with the vcpkg port, as well as getting KDE to use current libofx version to package its finance software.

**Comment by cstim** [comment]
By the way I'm still very open to accept a CMakeLists.txt file that builds the library and maybe also the tools. As the whole code hasn't changed for many years, it's also unlikely to change quickly now, so just adding the CMakeLists.txt in parallel wouldn't cause conflicts for now.

**Comment by wrobelda** [comment]
> By the way I'm still very open to accept a CMakeLists.txt file that builds the library and maybe also the tools. As the whole code hasn't changed for many years, it's also unlikely to change quickly now, so just adding the CMakeLists.txt in parallel wouldn't cause conflicts for now.

I did, in fact, start working on it today. We already had a working boilerplate that was hacked for KDE purposes for Windows only, but that needs cleaning up and amending for other the platforms, as well as adding support for tools. If that's of any help, here's a mildly cleaned-up version of it: https://github.com/wrobelda/libofx/tree/cmake

**Comment by wrobelda** [comment]
@cstim I made some significant progress, but frankly speaking, working on cmake build system while navigating around the existing automake scripting quirks is a bit painful. Would you be open to scraping autotools altogether, assuming the cmake build is complete and fully replaces the previous automake?

**Comment by cstim** [comment]
I'm open to scrapping the autotools infrastructure altogether, yes. However, when initially setting up the CMake system, is it necessary to edit the autotools part at all? I thought you can just act as if the other didn't exist, then commit both in parallel for maybe one version, then drop the old one in the next version?

**Comment by jralls** [comment]
That's what we did in GnuCash.

**Comment by wrobelda** [comment]
> I'm open to scrapping the autotools infrastructure altogether, yes. However, when initially setting up the CMake system, is it necessary to edit the autotools part at all? I thought you can just act as if the other didn't exist, then commit both in parallel for maybe one version, then drop the old one in the next version?

Ideally I'd do it this way, but there's some functionality supported by Autotools that I'd like to discuss whether it is worth keeping at all. Let's discuss the technical details in #63 draft PR I created.
